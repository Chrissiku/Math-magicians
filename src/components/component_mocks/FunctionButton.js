/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

class FunctionButton extends React.Component {
  render() {
    // eslint-disable-next-line react/prop-types
    return (
      <button type="button" id="clear" onClick={this.props.e}>
        {this.props.name}
      </button>
    );
  }
}

export default FunctionButton;
