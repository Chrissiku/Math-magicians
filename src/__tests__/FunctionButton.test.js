/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import FunctionButton from '../components/component_mocks/FunctionButton';

const props = { name: 'Chris', e: jest.fn() };

describe('Test Function Butoons', () => {
  test('Test Function Button component', () => {
    const TREE = renderer.create(<FunctionButton {...props} />).toJSON();
    expect(TREE).toMatchSnapshot();
  });

  it('Test the function button click', () => {
    render(<FunctionButton {...props} />);
    fireEvent.click(screen.getByText(props.name));
    expect(props.e).toHaveBeenCalledTimes(1);
  });
});
