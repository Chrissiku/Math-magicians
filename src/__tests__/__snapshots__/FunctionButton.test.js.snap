// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Function Butoons Test Function Button component 1`] = `
<button
  id="clear"
  onClick={[MockFunction]}
  type="button"
>
  Chris
</button>
`;
